channels/security/websocket.py

```diff
@@ -10,26 +12,38 @@ class OriginValidator:
     """
     Validates that the incoming connection has an Origin header that
     is in an allowed list.
+
+    If full attr is True, any origin from allowed_origins must be scheme://hostname[:port].
+    Port is optional, but recommended.
+    Check_cert works only with full attr.
     """
 
-    def __init__(self, application, allowed_origins):
+    def __init__(self, application, allowed_origins, full=False, check_cert=False):
```

Comment:
Can we have a better name for this than `full`?

Reply:
Can you suggest a better name than `` full``, maybe ``origin`` or ``scheme``. Is it necessary to leave the possibility of verifying the certificate or include it in the full check origin.